src/subcommands/diff.rs

```diff
@@ -24,12 +24,14 @@ pub fn diff(
         |f: &Path| f.starts_with("/proc/self/fd/") || f.starts_with("/dev/fd/");
 
     let diff_cmd = if via_process_substitution(minus_file) || via_process_substitution(plus_file) {
-        ["diff", "-u", "--"].as_slice()
+        format!("diff -u {} --", config.diff_args)
     } else {
```

Comment:
diff's `-u` (defaults to 3 context) will override a _following_ `-U<N>` (GNU diffutils at least), so replace with `-U3` which can be overwritten (and adapt help output above).

Reply:
OK, I will change this. For the record, I couldn't reproduce that on my Mac with `diff (GNU diffutils) 3.10` from homebrew. With that (and the `diff` that ships with Mac) `diff -u -U99` gives the same result as `diff -U3 -U99`.

Reply:
You're right, `-U3 -U4` works fine. Upon further inspection it seems that any previously set unified context can only be _increased_, so `-u` (3) then `-U4` works fine, but `-U3 -U1` or `-u -U1` will leave the context at 3, not set it to 1. The diff args will have to be scanned for any options that set the unified context :face_with_diagonal_mouth:

Reply:
Added detection of whether diff args imply that context is set by user and added tests.